### PR TITLE
[google-cloud-cpp] update to the latest release (v1.26.0)

### PR DESCRIPTION
**Pull request by coryan**
Update google-cloud-cpp to v1.16.0, the latest release.

- What does your PR fix?

N/A

- Which triplets are supported/not supported? Have you updated the CI baseline?

N/A

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

**Comment by coryan** [comment]
The `x64_linux` error seems unrelated (maybe a ci script problem?).

File: versions/baseline.json [hunk 1]
```diff
@@ -2281,8 +2281,8 @@
       "port-version": 4
     },
     "google-cloud-cpp": {
-      "baseline": "1.25.0",
-      "port-version": 3
+      "baseline": "1.26.0",
+      "port-version": 1
```
Comment (JonLiu1993) [inline versions/baseline.json:2285]: @coryan ，When the port is upgraded, we should set the port-version to 0 or delete

File: ports/google-cloud-cpp/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "google-cloud-cpp",
-  "version": "1.25.0",
-  "port-version": 3,
+  "version": "1.26.0",
+  "port-version": 1,
```
Comment (JonLiu1993) [inline ports/google-cloud-cpp/vcpkg.json:4]: @coryan ，When the port is upgraded, we should set the port-version to 0 or delete

Comment (coryan) [reply ports/google-cloud-cpp/vcpkg.json]: Fixed. Please take another look.

**Comment by ras0219-msft** [comment]
Thanks for the update!

**Pull request merged**